### DISCUSSION TITLE
WebGet: the help file does not adequately describe the capabilities of this function

### DISCUSSION THREAD

**Discussion by tozGitHub** [WinEdt-Team/WinEdt-Support#25] [1.1 General Help and Discussion]
Dear Alex, dear Community,

I played a bit around with the new WebGet. In my opinion, the help does not adequately describe the capabilities of this function. Just by reading the description, it was not clear to me whether we can only download files or also "webpages". Here are some things that I tried:
```
//Web pages
//WebGet("https://www.winedt.com/","%b\WE.html",1);
//Wikidata
//WebGet("https://www.wikidata.org/wiki/Q1755575","%b\data.txt",1);
//Maybe the earliest paper on arXiv
//(see https://dabacon.org/pontiff/2010/07/13/first-arxiv-paper/)
//WebGet("https://arxiv2bibtex.org/?q=hep-th/9108001&format=bibtex","%b\arxiv.txt",1);
//arXiv API
//WebGet("http://export.arxiv.org/api/query?search_query=lastUpdatedDate:[000012310900+TO+199108150000]&max_results=100&sortBy=lastUpdatedDate","%b\arxivApi.txt",1);
//Zentralblatt API
//WebGet("https://api.zbmath.org/v1/document/_search?search_string=An extension of Lomonosov's techniques&page=0&results_per_page=100","%b\Zentralblatt.txt",1);
//Wikipedia API
//WebGet("https://en.wikipedia.org/w/rest.php/v1/page/Earth","%b\wp.txt",1);
//Wiktionary API
WebGet("https://en.wiktionary.org/w/api.php?action=parse&page=cat&format=json","%b\Wikidata.txt",1);
IfOK("Prompt('OK');","Prompt('Not OK');");   
```
My attempts were successful.
We can also think of using some other APIs. I think that there is a lot of power behind WebGet. For example, it should be easy to download bibliographic information or some information from Wikidata or Wikipedia.
There is also the question whether there could be an extension (or a sister function) that does not download a file, but the content of a web page and saves it directly in a register without saving it as a file first. I think that the community could come up with a lot of creative uses of WebGet, which could be recorded partially as examples in the help file.

Best,
Thomas

**Comment by Alex-Simonic** [comment]
WinEdt uses this function sparingly in its default settings: check for updates and download extra samples. Users can explore and find different uses for it. A sister function can be simulated by adding `ReadFile(...)` and `DeleteFile(...)`. Previous versions of WinEdt implemented this functionality by launching PowerShell in the background. WebGet is faster and does not depend on the version of PowerShell but its presence may trigger security concerns for some users or organizations. That's why it is used sparingly in defaults...

**Comment by tozGitHub** [comment]
Dear Alex, dear community,

Alex, thank you for all your answers.
Maybe the lounge here on GitHub could be a place to share creative uses of WebGet.

Best,
Thomas
